support/ray_cluster_client.go

```diff
@@ -86,11 +115,51 @@ func (client *rayClusterClient) CreateJob(job *RayJobSetup) (response *RayJobRes
 	return
 }
 
+func (client *rayClusterClient) GetJobs() ([]map[string]interface{}, error) {
+	getAllJobsDetailsURL := client.endpoint.String() + "/api/jobs/"
+
+	req, err := http.NewRequest(http.MethodGet, getAllJobsDetailsURL, nil)
+	if err != nil {
+		return nil, err
+	}
+	if client.bearerToken != "" {
+		req.Header.Set("Authorization", "Bearer "+client.bearerToken)
+	}
+	resp, err := client.httpClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode == 503 {
+		return nil, fmt.Errorf("service unavailable")
+	}
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	var result []map[string]interface{}
```

Comment:
Is there any CustomResource which you can map the response to?
Returning CustomResource struct makes operations on it much easier.